### ISSUE TITLE
GPG signature for this Yarn release is invalid!

### ISSUE THREAD

**Issue by devshark** [yarnpkg/yarn#6591]
**Do you want to request a *feature* or report a *bug*?**
*bug*

**What is the current behavior?**
When I upgrade yarn using below, it shows this **BAD** error message. 
```sh
$ curl --compressed -o- -L https://yarnpkg.com/install.sh | bash
```
![yarn bug](https://user-images.githubusercontent.com/769245/47630309-ff44eb80-db7a-11e8-8747-d88bdd627110.png)

**If the current behavior is a bug, please provide the steps to reproduce.**
Same as above.
I have also tried to re-import the public GPG keys as mentioned in https://github.com/yarnpkg/yarn/issues/5468 but there is no change.
```bash
$ curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | gpg --import
gpg: key 86E50310: "Yarn Packaging <yarn@dan.cx>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
```

**What is the expected behavior?**
It should upgrade my yarn installation, without this serious red alert.

**Please mention your node.js, yarn and operating system version.**
Windows: 10
Node: 8.11.2
Yarn: 1.7.0

**Comment by fourthchakra** [comment]
I'm also experiencing this problem, running Mac OS X 10.14.1, node v8.6.0. The command apparently removed yarn, but the prior installed version is 1.12.1.

tdc@vajra ~/Projects/Facilitator $ curl --compressed -o- -L https://yarnpkg.com/install.sh | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2314  100  2314    0     0   1852      0  0:00:01  0:00:01 --:--:--  1852
Installing Yarn!
/Users/tdc/.yarn/bin/yarn
> 1.12.1 is already installed, Specified version: 1.12.3.
> Downloading tarball...

[1/2]: https://yarnpkg.com/latest.tar.gz --> /var/folders/g4/hr5yb0808xlbhm001s759b6r003142/T/yarn.tar.gz.XXXXXXXXXX.RWKPzyjy
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    93  100    93    0     0     22      0  0:00:04  0:00:04 --:--:--    22
100   609    0   609    0     0     60      0 --:--:--  0:00:10 --:--:--   182
  4 1139k    4 51786    0     0    162      0  2:00:00  0:05:18  1:54:42     0
curl: (56) LibreSSL SSL_read: SSL_ERROR_SYSCALL, errno 54

[2/2]: https://yarnpkg.com/latest.tar.gz.asc --> /var/folders/g4/hr5yb0808xlbhm001s759b6r003142/T/yarn.tar.gz.XXXXXXXXXX.RWKPzyjy.asc
100    97  100    97    0     0   1526      0 --:--:-- --:--:-- --:--:--  1526
100   613    0   613    0     0    347      0 --:--:--  0:00:01 --:--:--   740
100   832  100   832    0     0    167      0  0:00:04  0:00:04 --:--:--   270
> Verifying integrity...
gpg: Signature made Wed Nov  7 07:13:31 2018 PST
gpg:                using RSA key E21930C4D0A4AA4618581F7AE074D16EB6FF4DE3
gpg: BAD signature from "Yarn Packaging <yarn@dan.cx>" [unknown]
> GPG signature for this Yarn release is invalid! This is BAD and may mean the release has been tampered with. It is strongly recommended that you report this to the Yarn developers.
tdc@vajra ~/Projects/Facilitator $ node -v
v8.6.0
tdc@vajra ~/Projects/Facilitator $ yarn -v
-bash: /Users/tdc/.yarn/bin/yarn: No such file or directory

**Comment by Daniel15** [comment]
Yeah the script doesn't handle cURL failures very well at the moment. There's an  issue for that already.

@devshark Your particular issue looks like a bad download - The tarball was only 349 bytes.

**Comment by jhabdas** [comment]
My terminal when verifying the latest version (today):

<pre>
> Verifying integrity...
gpg: key 1646B01B86E50310: "Yarn Packaging <yarn@dan.cx>" 2 new signatures
gpg: key 1646B01B86E50310: "Yarn Packaging <yarn@dan.cx>" 2 new subkeys
gpg: Total number processed: 1
gpg:            new subkeys: 2
gpg:         new signatures: 2
gpg: Signature made Wed Dec 19 00:04:55 2018 WITA
gpg:                using RSA key E21930C4D0A4AA4618581F7AE074D16EB6FF4DE3
gpg: Good signature from "Yarn Packaging <yarn@dan.cx>" [unknown]
gpg: Note: This key has expired!
</pre>

Yarn:

> GPG signature looks good

What's the sense of having an integrity check if the app which does the checking is the one which could be suspicious?

**Comment by mateusz-gozdek-sociomantic** [comment]
So we hit the issue as well. It seems that the key was refreshed correctly, so now you need to execute either `apt-key adv --keyserver keyserver.ubuntu.com --refresh-keys` on Ubuntu or `gpg --refresh-keys` to fetch refreshed key, then it all works.
